sequence.go

```diff
@@ -92,3 +92,77 @@ func getFeatureSequence(feature Feature, location Location) string {
 
 	return sequenceString
 }
+
+func cartRune(inList ...[]rune) [][]rune {
+	// An iteratitive approach to calculate Cartesian product of two or more lists
+	// Adapted from https://rosettacode.org/wiki/Cartesian_product_of_two_or_more_lists
+	// supposedly "minimizes allocations and computes and fills the result sequentially"
+
+	var possibleVariants int = 1 // a counter used to determine the possible number of variants
+	for _, inList := range inList {
+		possibleVariants *= len(inList)
+	}
+	if possibleVariants == 0 {
+		return nil // in the future this could be part of an error return?
+	}
+	allVariants := make([][]rune, possibleVariants)              // this is the 2D slice where all variants will be stored
+	variantHolders := make([]rune, possibleVariants*len(inList)) // this is an empty slice with a length totaling the size of all input characters
+	variantChoices := make([]int, len(inList))                   // these will be all the possible variants
+	start := 0
+	for variant := range allVariants {
+		end := start + len(inList) // define end point
+		variantHolder := variantHolders[start:end]
+
+		allVariants[variant] = variantHolder
+
+		start = end // start at end point
+
+		for variantChoicesIndex, variantChoice := range variantChoices {
+			variantHolder[variantChoicesIndex] = inList[variantChoicesIndex][variantChoice]
+		}
+		for variantChoicesIndex := len(variantChoices) - 1; variantChoicesIndex >= 0; variantChoicesIndex-- {
+			variantChoices[variantChoicesIndex]++
+			if variantChoices[variantChoicesIndex] < len(inList[variantChoicesIndex]) {
+				break
+			}
+			variantChoices[variantChoicesIndex] = 0
+		}
+	}
+	return allVariants
+}
+
+func AllVariantsIUPAC(seq string) []string {
+	//AllVariantsIUPAC( seq string) takes a string as input
+	// and returns all iupac variants as output
+	seqVariantList := [][]rune{}
+	seqVariants := []string{}
```

Comment:
If there is only the option for uppercase runes, the sequence should be set to uppercase before it is used in the seqVariantList.

Reply:
This has now been resolved!